### PR TITLE
Suggest adding/removing `ref` for binding patterns

### PR DESCRIPTION
**Pull request by TaKO8Ki**
This fixes what a fixme comment says.

r? @estebank

### CODE REVIEW

**Comment by joshtriplett** [comment]
LGTM

File: compiler/rustc_typeck/src/check/pat.rs [hunk 1]
```diff
@@ -645,12 +652,50 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
+                    "consider adding `ref`",
+                    "ref ",
+                    Applicability::MaybeIncorrect,
+                );
+            }
+            (_, ty::Ref(_, inner_ty, _), hir::BindingAnnotation::Ref)
+                if self.can_eq(self.param_env, expected, *inner_ty).is_ok() =>
+            {
+                err.span_suggestion_verbose(
+                    span.with_hi(span.lo() + BytePos(4)),
```
Comment (estebank) [inline compiler/rustc_typeck/src/check/pat.rs:689]: We should avoid this kind of span building when possible.

**Pull request merged**